ludwig/data/dataset/ray.py

```diff
@@ -30,6 +30,7 @@
 from ray.data.dataset_pipeline import DatasetPipeline
```

Comment:
An alternative way to implement the changes in this file would be to update `RayDatasetBatcher._fetch_next_batch` method. Specifically, in (this line)[https://github.com/ludwig-ai/ludwig/blob/2b81fcfc8c0329ed81ec723235cfd71384f70135/ludwig/data/dataset/ray.py#L339], you can add a check to raise `StopIteration` if the batch only has a single item.

This will allow you to remove the `_set_ignore_last` function, and have no need to pass in the `ignore_last` argument in `iter_batches`. Wdyt?

Reply:
Updated!